packages/ibm-products/src/components/UserAvatar/UserAvatar.test.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react'; // https://testing-library.com/docs/react-testing-library/intro
+
+import { pkg } from '../../settings';
+import uuidv4 from '../../global/js/utils/uuidv4';
+
+import { UserAvatar } from '.';
+
+const blockClass = `${pkg.prefix}--user-avatar`;
+const componentName = UserAvatar.displayName;
+
+// values to use
+const dataTestId = uuidv4();
+
+const renderComponent = ({ ...rest } = {}) =>
+  render(<UserAvatar {...{ ...rest }} />);
+
+describe(componentName, () => {
+  it('should return a circle with background color', async () => {
+    const { container } = renderComponent({
+      backgroundColor: 'light-cyan',
+    });
+    const elements = container.getElementsByClassName(blockClass);
+    const element = elements.length > 0 ? elements[0] : null;
```

Comment:
If we update `UserAvatar` to use `role="img"`, then I think you could simplify this to ...

```suggestion
    renderComponent({
      backgroundColor: 'light-cyan',
    });
    const element = screen.getByRole('img');
```

I believe Matt’s original comment was about querying elements for testing[^1][^2].

[^1]: [Common mistakes with React Testing Library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-container-to-query-for-elements) by Kent C. Dodds
[^2]: [About Queries](https://testing-library.com/docs/queries/about#priority) from Testing Library docs